apps/website/src/pages/insights/epics/[epic].tsx

```diff
@@ -87,7 +85,60 @@ const EpicsDetailPage: Page<Props> = props => {
     {
       // Prevent animation if we go out of the page
       refetchOnWindowFocus: false,
-      initialData: props.burnup,
+      onSuccess: data => {
+        const differenceBetweenSelectedDates = differenceInCalendarDays(
+          selectedDates?.to || new Date(),
+          selectedDates?.from || new Date()
+        )
+
+        const rate = 50 // 1 sample per 50 days
+
+        let samplingRate = rate // Use the default rate as the initial value
+
+        if (differenceBetweenSelectedDates > 0) {
+          // Calculate the total number of data points within the selected date range
+          const totalDataPoints = data?.gh_burnup.length || 0
+
+          // Calculate the desired number of data points based on the sampling rate
+          const desiredDataPoints = Math.ceil(totalDataPoints / rate)
+
+          // Calculate the actual sampling rate based on the desired number of data points
+          samplingRate = Math.max(
+            1,
+            Math.floor(totalDataPoints / desiredDataPoints)
+          )
+        }
+
+        // Downsampling the burnup data
+        const downsampledData: GetBurnupQuery['gh_burnup'] = []
+
+        if (data?.gh_burnup.length > 0) {
+          data?.gh_burnup.forEach((dataPoint, index) => {
+            if (index % samplingRate === 0) {
+              downsampledData.push(dataPoint)
+            }
+          })
+        }
+
+        if (
+          selectedDates?.from &&
+          selectedDates?.to &&
+          data?.gh_burnup.length === 0
```

Comment:
This state can be derived instead of creating another state to sync. Can be done in the follow up.